src/lib/features/feature-search/feature-search-store.ts

```diff
@@ -326,6 +327,43 @@ class FeatureSearchStore implements IFeatureSearchStore {
                 'ranked_features.feature_name',
                 'lifecycle.stage_feature',
             );
+        if (this.flagResolver.isEnabled('flagsOverviewSearch')) {
+            finalQuery
```

Comment:
we add it as late as possible so that we operate on pre-filtered features. We find all change request events for a give feature and then find which of those change request events have a corresponding change requests in a given environment.

Reply:
I was considering moving the code into a new method but other subqueries are inlined so it's consistent and also I'm not sure if jumping to the extracted method will help a lot